element_facemap/facial_behavior_estimation.py

```diff
@@ -301,33 +300,37 @@ def make(self, key):
             # update processing_output_dir
             FacemapTask.update1({**key, "facemap_output_dir": output_dir.as_posix()})
 
+        output_dir = find_full_path(get_facemap_root_data_dir(), output_dir)
+
         if task_mode == "trigger":
             from facemap.process import run as facemap_run
 
             params = (FacemapTask & key).fetch1("facemap_params")
 
+            valid_args = inspect.getfullargspec(facemap_run).args
+            params = {k: v for k, v in params.items() if k in valid_args}
+
             video_files = (FacemapTask * VideoRecording.File & key).fetch("file_path")
+            # video files are sequentially acquired, not simultaneously
             video_files = [
-                [
-                    find_full_path(get_facemap_root_data_dir(), video_file).as_posix()
-                    for video_file in video_files
-                ]
+                [find_full_path(get_facemap_root_data_dir(), video_file).as_posix()]
+                for video_file in video_files
             ]
 
-            output_dir = find_full_path(get_facemap_root_data_dir(), output_dir)
-            facemap_run(
-                video_files,
-                sbin=params["sbin"],
-                proc=params,
-                savepath=output_dir.as_posix(),
-                motSVD=params["motSVD"],
-                movSVD=params["movSVD"],
-            )
+            @memoized_result(uniqueness_dict=params, output_directory=output_dir)
+            def run_facemap_process():
+                facemap_run(
+                    filenames=video_files,
+                    savepath=output_dir.as_posix(),
+                    **params,
+                )
 
-        _, creation_time = get_loader_result(key, FacemapTask)
-        key = {**key, "processing_time": creation_time}
+            run_facemap_process()
```

Comment:
I would suggest to make the function private and rename it to `_run_facemap_process()`.

Reply:
An example [here](https://github.com/datajoint/element-array-ephys/blob/b510cb8aa5499d3f66adae1aa9a168c25303ac19/element_array_ephys/spike_sorting/si_spike_sorting.py#L208-L224), `_run_sorter()` in `element-array-ephys`.